src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/ReflectionJsonTypeInfoOfT.cs

```diff
@@ -166,6 +166,7 @@ private void AddPropertiesAndParametersUsingReflection()
             }
         }
 
+        [RequiresDynamicCode(JsonSerializer.SerializationRequiresDynamicCodeMessage)]
```

Comment:
this is not needed considering constructors are already annotated. better if we suppress this